### DISCUSSION TITLE
Increase in PutLogEvents for cloud watch due to batch job

### DISCUSSION THREAD

**Discussion by lacoak21** [nextflow-io/nextflow#5370] [Q&A]
I am trying to disable cloud watch logging because we had a multifold increase in Cloud Watch PutLogEvents. We didn't change anything for our cextflow pipelines and configuration so we are still unclear what happened. It was confirmed that the increase came from nextflow tower resources. In the meantime I would like to switch from forwarding logs to CloudWatch to 
"logConfiguration": {
      "logDriver": "json-file",
      "options": {},
      "secretOptions": []
    },

It seems like there is no easy way to do this since next flow automatically creates the job definitions.

I could manually update the job definitions through the AWS-cli and then use process.container = 'job-definition://your-job-definition-name' for each one but we have many and that would be a tedious solution. 

Do you have any suggestions?

**Comment by bentsherman** [comment]
Maybe the `aws.batch.logsGroup` config option?

https://nextflow.io/docs/latest/reference/config.html#aws

**Comment by mearatjames** [comment]
@bentsherman I'm having similar issue to @lacoak21 where I would like to disable the cloudwatch logs for the nextflow run using aws-batch to save on the aws cost. 

I tried by removing the PutLogEvents permission from the Compute Environment instance role (based on what I found here https://nextflow.slack.com/archives/C02T97HAV5M/p1666983382733349?thread_ts=1664982143.297759&cid=C02T97HAV5M). However that resulted in the run failing because it cannot find the cloudwatch log for the jobs.

I tried looking into the aws.batch.logsGroup config option, but that seems to only allow to change the name of the cloudWatch logsGroup.

Is there a way that I can disable the cloudwatch logs for the nextflow run using aws-batch? Any suggestion is greatly appreciated.

**Comment by jvolkening** [comment]
Bump... Has anyone figured out a method to achieve this (disabling CloudWatch logs for `awsbatch` executor)? It is somewhat critical for us to figure this out, as we just (for the second time) incurred significant AWS CloudWatch charges due to a rogue process dumping massive amounts of text to STDERR. We can fix the specific cases as they arise, but it would provide significant peace of mind to find a global solution.

Any guidance would be greatly appreciated.

**Reply by bentsherman** [reply]
Unfortunately I'm not very familiar with CloudWatch configuration. If you can find documentation on configuring/disabling CloudWatch when submitting an AWS Batch job, we could try to expose that as a Nextflow config option

**Comment by jvolkening** [comment]
There is a lot of conflicting information floating around, but it seems like the most promising possibility that is actually documented by AWS would be to expose the `containerProperties`/`logConfiguration`/`logDriver` parameter as a NF config option. It looks like it is hardcoded to `awslogs` right now, which is the option for CloudWatch Logs:
https://github.com/nextflow-io/nextflow/blob/fd6012b520e8878d395b207ca9b0ec4eeae0859f/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy#L649
AWS doesn't allow the driver to be disabled completely via this setting, but it would allow the user to change it to `syslog` or another local option that would avoid the potential for unexpected CloudWatch charges.
